docs/data-sources/users.md

```diff
@@ -0,0 +1,80 @@
+---
+subcategory: "Security"
+---
+
+# databricks_users Data Source
+
+-> If you have a fully automated setup with workspaces created by [databricks_mws_workspaces](../resources/mws_workspaces.md) or [azurerm_databricks_workspace](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/databricks_workspace), please make sure to add [depends_on attribute](../guides/troubleshooting.md#data-resources-and-authentication-is-not-configured-errors) in order to prevent _default auth: cannot configure default credentials_ errors.
+
+Retrieves information about multiple [databricks_user](../resources/user.md) resources.
+
+## Example Usage
+
+Adding a subset of users to a group
+
+```hcl
+data "databricks_users" "company_users" {
+  filter = "userName co \"@domain.org\""
+}
+
+resource "databricks_group" "data_users_group" {
+  display_name = "Data Users"
+}
+
+resource "databricks_group_member" "add_users_to_group" {
+  for_each = { for user in data.databricks_users.company_users.users : user.id => user }
+  group_id  = databricks_group.data_users_group.id
+  member_id = each.value.id
+}
+```
+
+## Argument Reference
+
+This data source allows you to filter the list of users using the following optional arguments: 
+
+- `filter` - (Optional) Query by which the results have to be filtered. If not specified, all users will be returned. Supported operators are equals (`eq`), contains (`co`), starts with (`sw`), and not equals (`ne`). Additionally, simple expressions can be formed using logical operators `and` and `or`.
```

Comment:
Maybe we should add a note (or pointer to the doc) that will clarify if the search is case-sensitive or not?